src/wallet/wallet.cpp

```diff
@@ -226,6 +228,41 @@ CKeyID Wallet::CreateNewKey(bool compressed) {
     return addr;
 }
 
+std::vector<CKeyID> Wallet::GetAllAddresses() {
+    std::vector<CKeyID> result;
+    result.reserve(keyBook.size());
+
+    for (auto it = keyBook.begin(); it != keyBook.end(); it++) {
+        result.push_back(it->first);
+    }
+
+    return result;
+}
+
+std::unordered_map<std::string, std::vector<TxOutput>> Wallet::GetTxoutsWithAddr() {
+    std::unordered_map<std::string, std::vector<TxOutput>> mAddr2Outputs;
+
+    auto decodedAddrs = GetAllAddresses();
+    for (const auto& addr : decodedAddrs) {
+        mAddr2Outputs.emplace(EncodeAddress(addr), std::vector<TxOutput>{});
+    }
+
+    for (const auto& keypair : unspent) {
+        const auto addr  = EncodeAddress(std::get<0>(keypair.second));
+        const auto entry = mAddr2Outputs.find(addr);
+        if (entry != mAddr2Outputs.end()) {
```

Comment:
Is it necessary to check the address? I suggest output all the addresses and tx outputs.